### PR TITLE
Introducing a very lightweight Statistics screen.

### PR DESCRIPTION
**Pull request by Dorvaryn**
In our quest to get all features from the Google sample into this app, here is the statistics screen.

We access it through the overflow menu at the moment. The nav drawer can be implemented as a separate PR.

##### Screenshots
| Desired | After |
| ----- | ----- |
|device-2016-07-07-114130|device-2016-07-07-150146|
|device-2016-07-07-114632|device-2016-07-07-150120|

### CODE REVIEW

File: Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/statistics/presenter/StatisticsPresenterTest.kt [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    }
+
+    private fun statistics() = Statistics.builder().activeTasks(1).completedTasks(3).build()
+
+    private fun setUpService() {
+        statisticsEventSubject = BehaviorSubject.create()
+        Mockito.`when`(service.getStatisticsEvent()).thenReturn(statisticsEventSubject)
+    }
+
+}
```
Comment (joetimmins) [inline Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/statistics/presenter/StatisticsPresenterTest.kt]: Good concise tests! 👍

File: Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/statistics/service/TasksStatisticsServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    private fun setUpService() {
+        tasksEventSubject = BehaviorSubject.create()
+        val tasksEventReplay = tasksEventSubject.replay()
+        tasksEventReplay.connect()
+        Mockito.`when`(tasksService.getTasksEvent()).thenReturn(tasksEventReplay)
+    }
+
+    private fun emptyStatistics() = Statistics.builder().build()
+
+    private fun sampleStatisticsSomeCompletedSomeActiveTasks() = Statistics.builder()
```
Comment (joetimmins) [inline Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/statistics/service/TasksStatisticsServiceTest.kt]: `statisticsWithSomeActiveAndSomeCompletedTasks` ?

File: Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/statistics/service/TasksStatisticsServiceTest.kt [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            .activeTasks(1)
+            .completedTasks(3)
+            .build()
+
+    private fun sampleSomeCompletedSomeActiveTasks() = Tasks.asSynced(listOf(
```
Comment (joetimmins) [inline Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/statistics/service/TasksStatisticsServiceTest.kt]: `tasksWithSomeActiveAndSomeCompleted` ? Might just read a little easier, i don't think `sample` is adding much here

**Comment by joetimmins** [comment]
LGTM 👍

File: Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/statistics/presenter/StatisticsPresenterTest.kt [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    private fun givenThePresenterIsPresenting() {
+        presenter.startPresenting()
+    }
+
+    private fun givenThePresenterIsNotPresenting() {
+    }
+
+    private fun statistics() = Statistics.builder().activeTasks(1).completedTasks(3).build()
+
+    private fun setUpService() {
```
Comment (frapontillo) [inline Architecture/TodoApp/android/core/src/test/kotlin/com/novoda/todoapp/statistics/presenter/StatisticsPresenterTest.kt]: Maybe re-arrange this private methods in a top-down fashion? Don't mind leaving them like this if it's your standard here.

**Pull request merged**